src/mig/modules/migoval/migoval.go

```diff
@@ -0,0 +1,227 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+package migoval
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	ovallib "github.com/ameihm0912/mozoval/go/src/oval"
+	"io"
+	"io/ioutil"
+	"mig/modules"
+	"time"
+)
+
+var stats Statistics
+
+// Various counters used to populate module statistics at the end of the
+// run.
+var counters struct {
+	startTime time.Time
+}
+
+func startCounters() {
+	counters.startTime = time.Now()
+}
+
+func endCounters() {
+	stats.OvalRuntime = time.Now().Sub(counters.startTime)
+}
+
+func init() {
+	modules.Register("oval", func() interface{} {
+		return new(Runner)
+	})
+}
+
+type Runner struct {
+	Parameters Parameters
+	Results    modules.Result
+}
+
+func (r Runner) Run(in io.Reader) (resStr string) {
+	defer func() {
+		if e := recover(); e != nil {
+			// return error in json
+			r.Results.Errors = append(r.Results.Errors, fmt.Sprintf("%v", e))
+			r.Results.Success = false
+			endCounters()
+			r.Results.Statistics = stats
+			err, _ := json.Marshal(r.Results)
+			resStr = string(err)
+			return
+		}
+	}()
+
+	startCounters()
+
+	// Read module parameters from stdin
+	err := modules.ReadInputParameters(in, &r.Parameters)
+	if err != nil {
+		panic(err)
+	}
+
+	err = r.ValidateParameters()
+	if err != nil {
+		panic(err)
+	}
+
+	ovallib.Init()
+	ovallib.SetMaxChecks(r.Parameters.MaxConcurrentEval)
```

Comment:
Do you need concurrency because of blocking operations, or horsepower?
If the former, what about limiting the max number of CPUs that can be used to one or two, and let the lib use more goroutine without using more cpus.
```go
	runtime.GOMAXPROCS(1)
```